project-api/project-api-test/src/test/java/io/ballerina/projects/test/PackageResolutionTests.java

```diff
@@ -268,6 +274,107 @@ public void testProjectSaveWithNoWritePermission() throws IOException {
+
+        // - Step 4.1 : Get the main.bal file document for ProjectA
+        String newMainProjectAContent = "";
+        Module defaultModuleProjectA = loadProjectA.currentPackage().getDefaultModule();
+        Optional<DocumentId> mainDocumentId =
+                defaultModuleProjectA.documentIds()
+                        .stream()
+                        .filter(documentId -> defaultModuleProjectA.document(documentId).name().equals("main.bal"))
+                        .findFirst();
+        if (mainDocumentId.isEmpty()) {
+            Assert.fail("Failed to retrieve the document ID");
+        }
+
+        // - Step 4.2 : Modify the content of ProjectA
+        Document document = defaultModuleProjectA.document(mainDocumentId.get());
+        document.modify().withContent(newMainProjectAContent).apply();
+
+
+        // Step 5 : Compile ProjectA and verify dependency
+        compilation = loadProjectA.currentPackage().getCompilation();
+        dependencyGraph = compilation.getResolution().dependencyGraph();
+        for (ResolvedPackageDependency graphNode : dependencyGraph.getNodes()) {
+            PackageManifest manifest = graphNode.packageInstance().manifest();
+            if (manifest.name().value().equals("projectB")) {
+                Assert.fail("ProjectB found in dependency after editing ProjectA");
+            }
+        }
+
+
+        // Step 6 : Clear ProjectB cache
+        FileUtils.deleteDirectory(projectBCachePath.toFile());
+
+
+        // Step 7 : Compile ProjectB2 and cache
+        depCompileResult = BCompileUtil.compileAndCacheBala("projects_for_resolution_tests/projectB2");
+        if (depCompileResult.getErrorCount() > 0) {
+            Assert.fail("Package B contains compilations error");
+        }
+
+
+        // Step 8 : Modify ProjectA again with the old content
+
+        // - Step 8.1 : Get the main.bal file document
+        String oldMainProjectAContent = "import samjs/projectB;\n" + "\n" + "public function getHello() returns " +
+                "(string) {\n" + "    return projectB:hello();\n" + "}";
+
+        Module oldModuleProjectA = loadProjectA.currentPackage().getDefaultModule();
+
+        mainDocumentId = defaultModuleProjectA.documentIds()
+                .stream()
+                .filter(documentId -> oldModuleProjectA.document(documentId).name().equals("main.bal"))
+                .findFirst();
+        if (mainDocumentId.isEmpty()) {
+            Assert.fail("Failed to retrieve the document ID");
+        }
+
+        // - Step 8.2 : Modify the content
+        document = defaultModuleProjectA.document(mainDocumentId.get());
+        document.modify().withContent(oldMainProjectAContent).apply();
+
+
+        // Step 9 : Compile ProjectA and verify dependency
+        compilation = loadProjectA.currentPackage().getCompilation();
+        Assert.assertFalse(compilation.diagnosticResult().errorCount() == 0, "Package A has compiled successfully " +
```

Comment:
We can simplify this assertion to use AssertNotEquals

Reply:
Made use of AssertNotEquals